### PR TITLE
Improve WETH page

### PR DESCRIPTION
**Pull request by fabioberger**
This PR:
* Adds hack to Portal such that it acts **as if** new WETH token is already in TokenRegistry without that being the case. We are waiting on a couple dependent parties before updating the tokenRegistry.
* Adds a fill **Max** amount button to the WETH unwrap dialog.
* Adds a notice dialog to Portal balances section notifying users that we've upgraded to a new WETH contract and informing them of the new section where they can unwrap ETH held by the outdated contract.
* Adds address tooltips and links to Etherscan to all WETH tokens in the `Wrap ETH` section.
* Fixed bug such that `Wrap ETH` page loads on networks without any outdated WETH contracts.
* Fixed bug where if reporting error takes too long, UI isn't showing user error immediately.

screen shot 2017-12-18 at 09 29 35
- This dialog shows up the first and only first time they visit the `balances` section.

### CODE REVIEW

File: packages/website/ts/utils/configs.ts [hunk 1]
```diff
@@ -19,6 +19,13 @@ export const configs = {
     lastLocalStorageFillClearanceDate: '2017-11-22',
     lastLocalStorageTrackedTokenClearanceDate: '2017-12-13',
     isMainnetEnabled: true,
+    shouldDeprecateOldWethToken: true,
+    // newWrappedEthers is temporary until we remove the shouldDeprecateOldWethToken flag
+    // and add the new WETHs to the tokenRegistry
+    newWrappedEthers: {
+        1: '0xe495bcacaf29a0eb00fb67b86e9cd2a994dd55d8',
+        42: '0x739e78d6bebbdf24105a5145fa04436589d1cbd9',
```
Comment (LogvinovLeon) [inline packages/website/ts/utils/configs.ts]: What about rinkeby?

Comment (fabioberger) [reply packages/website/ts/utils/configs.ts]: We haven't deployed to Rinkby yet, added it to this up-coming sprint though!

File: packages/website/ts/blockchain.ts [hunk 2]
```diff
@@ -158,6 +158,12 @@ export class Blockchain {
     }
     public async isAddressInTokenRegistryAsync(tokenAddress: string): Promise<boolean> {
         utils.assert(!_.isUndefined(this.zeroEx), 'ZeroEx must be instantiated.');
+        // HACK: temporarily whitelist the new WETH token address `as if` they were
```
Comment (BMillman19) [inline packages/website/ts/blockchain.ts]: lets make a corresponding task to remove the hack and link the ID here

File: packages/website/ts/blockchain.ts [hunk 3]
```diff
@@ -598,12 +604,11 @@ export class Blockchain {
             // new canonical WETH.
             // TODO: Remove this hack once we've updated the TokenRegistries
```
Comment (BMillman19) [inline packages/website/ts/blockchain.ts]: same here

File: packages/website/ts/components/dialogs/wrapped_eth_section_notice_dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+export function WrappedEthSectionNoticeDialog(props: WrappedEthSectionNoticeDialogProps) {
+    return (
+        <Dialog
+            title="Dedicated Wrapped Ether Section"
+            titleStyle={{fontWeight: 100}}
+            actions={[
+                <FlatButton
+                    key="acknowledgeWrapEthSection"
+                    label="Sounds good"
+                    onTouchTap={props.onToggleDialog.bind(this)}
```
Comment (BMillman19) [inline packages/website/ts/components/dialogs/wrapped_eth_section_notice_dialog.tsx]: do we want to bind `this` here? I can see someone passing a function here and expecting `this` to refer to to a different context

File: packages/website/ts/components/dialogs/wrapped_eth_section_notice_dialog.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+                />,
+            ]}
+            open={props.isOpen}
+            onRequestClose={props.onToggleDialog.bind(this)}
```
Comment (BMillman19) [inline packages/website/ts/components/dialogs/wrapped_eth_section_notice_dialog.tsx]: same here

File: packages/website/ts/components/dialogs/wrapped_eth_section_notice_dialog.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+            autoScrollBodyContent={true}
+            modal={true}
+        >
+            <div className="pt2" style={{color: colors.grey700}}>
+                <div>
+                    We have recently updated the Wrapped Ether token used by 0x Portal.
```
Comment (BMillman19) [inline packages/website/ts/components/dialogs/wrapped_eth_section_notice_dialog.tsx]: maybe add WETH in parentheses? ex. "We have recently updated the Wrapped Ether token (WETH) used by 0x Portal."

File: packages/website/ts/components/eth_wrappers.tsx [hunk 7]
```diff
@@ -345,9 +383,14 @@ export class EthWrappers extends React.Component<EthWrappersProps, EthWrappersSt
         });
     }
     private getOutdatedWETHAddresses(): string[] {
-        const outdatedWETHAddresses = _.map(configs.outdatedWrappedEthers, outdatedWrappedEther => {
-            return outdatedWrappedEther[this.props.networkId].address;
-        });
+        const outdatedWETHAddresses = _.compact(_.map(configs.outdatedWrappedEthers, outdatedWrappedEtherByNetwork => {
+            const  outdatedWrappedEtherIfExists = outdatedWrappedEtherByNetwork[this.props.networkId];
```
Comment (BMillman19) [inline packages/website/ts/components/eth_wrappers.tsx]: there's an extra space after const here

File: packages/website/ts/components/dialogs/wrapped_eth_section_notice_dialog.tsx (hunk 4 above)
Comment (fabioberger) [reply packages/website/ts/components/dialogs/wrapped_eth_section_notice_dialog.tsx]: Yep, 100% correct. Good catch.

File: packages/website/ts/components/eth_wrappers.tsx (hunk 7 above)
Comment (fabioberger) [reply packages/website/ts/components/eth_wrappers.tsx]: Huh, weird the linter isn't catching that. Any ideas @LogvinovLeon?

**Pull request merged**